apps/train_loc/lib/train_loc/vehicles/new_vehicle.ex

```diff
@@ -0,0 +1,18 @@
+defmodule TrainLoc.Vehicles.NewVehicle do
```

Comment:
question: this looks to be the same as `Vehicle`, but without `block`? Could they be combined with `block` becoming optional?

Reply:
I think it's possible, yes. For one thing I could just build structs without using `Vehicle.from_json`. With regards to `from_json`, there are couple differences are in Trip ID handling (they are strings in the new format, apparently integers in the old) and block IDs. I was wary to modify these but I'm fine doing it if it makes sense to you.